File: src/main/java/com/back/together02be/stock/repository/StockRepository.java
package com.back.together02be.stock.repository;

import java.util.Optional;
import org.springframework.data.jpa.repository.JpaRepository;
import com.back.together02be.stock.entity.Stock;
import java.util.List;

public interface StockRepository extends JpaRepository<Stock, Long> {

    List<Stock> findAll();

	Optional<Stock> findByStockCode(String stockCode);

	boolean existsByStockCode(String stockCode);

}

File: src/main/java/com/back/together02be/stock/repository/StockRepository.kt
package com.back.together02be.stock.repository

import com.back.together02be.stock.entity.Stock
import org.springframework.data.jpa.repository.JpaRepository

interface StockRepository : JpaRepository<Stock, Long> {

    override fun findAll(): List<Stock>

    // todo optional 확장함수로 처리
    fun findByStockCode(stockCode: String): Stock?

    fun existsByStockCode(stockCode: String): Boolean
}
